app/src/main/java/edu/cascadia/mobile/apps/layoutexamples/MainActivity.java

```diff
@@ -21,23 +22,27 @@ protected void onCreate(Bundle savedInstanceState) {
         CheckBox lemonCheck = findViewById(R.id.lemonBox);
 
         //TODO: Assign each checkbox an onClick EventHandler that calls the onCheckboxClicked method
-
+        milkCheck.setOnClickListener(v -> onCheckboxClicked(v));
+        sugarCheck.setOnClickListener(v -> onCheckboxClicked(v));
+        lemonCheck.setOnClickListener(v -> onCheckboxClicked(v));
     }
 
     public void onCheckboxClicked(View view) {
-        CheckBox clicked = (CheckBox)view;
+        CheckBox clicked = (CheckBox) view;
         //Add or remove the checkboxes label to the ArrayList
-        if(clicked.isChecked())
+        if (clicked.isChecked())
             TeaGoodies.add(clicked.getText());
         else
             TeaGoodies.remove(clicked.getText());
 
         //Create a String of the current set of labels
         StringBuilder teaTime = new StringBuilder("Tea with ");
-        for (CharSequence goodies :TeaGoodies) {
-            teaTime.append(goodies).append((", "));
-        }
+        for (CharSequence goodies : TeaGoodies) teaTime.append(goodies).append((", "));
         //Display the string as a temporary popup - called a "Toast"
         Toast.makeText(MainActivity.this, teaTime, Toast.LENGTH_SHORT).show();
     }
-}
+        public void onClick(View view) {
+            onCheckboxClicked(view);
```

Comment:
This will try to cast a button as checkbox, when the method is called (pass it null instead)